proposals/lip-add_weights_to_lisk-bft_consensus_protocol.md

```diff
@@ -0,0 +1,314 @@
+
+Below you find a list of the main functions in [LIP 0014][lisk-bft-lip] that define the computation of prevotes, precommits and properties `maxHeightPrevoted` and `maxHeightFinalized` and the corresponding function in this LIP.
+
+* The function `getHeightNotPrevoted` in [LIP 0014][lisk-bft-lip-comp] is the same in this LIP.
+* The function `applyPrevotesPrecommits` in [LIP 0014][lisk-bft-lip-comp] corresponds to the function `updatePrevotesPrecommits` in this LIP.
+* The function `computeChainMaxHeightPrevoted` in [LIP 0014][lisk-bft-lip-comp] corresponds to the function `updateMaxHeightPrevoted` in this LIP.
+* The function `computeChainMaxHeightFinalized` in [LIP 0014][lisk-bft-lip-comp] corresponds to the function `updateMaxHeightPrecommitted` in this LIP.
+
+The changes in these functions are due to the fact that all prevotes and precommits have an associated weight as described above and to account for the state structure defined in the [BFT module LIP][bft-module-lip].
+
+### Comparison to the Lisk-BFT paper
+
+The main technical difference between the Lisk-BFT protocol specified in the paper (Definition 4.1 in the [Lisk-BFT paper][lisk-bft-paper]) and the specification in this LIP is that the BFT weights and thresholds in this LIP are integers.
+For simplicity of exposition, the paper assumes that the BFT weights are scaled such that the sum of BFT weights is `1` at every height.
+To avoid any issues with floating point arithmetic and rounding, the LIP, however, uses integral weights instead.
+A transformation from the BFT weights and thresholds in this LIP to the BFT weights and thresholds used in the paper is done by dividing all BFT weights and thresholds at height `h` by the sum of BFT weights at height `h`.
+
+The Lisk-BFT protocol introduced in Definition 4.1 of the [Lisk-BFT paper][lisk-bft-paper] allows to choose two protocol parameters, `τ ∈ (1/3, 1]` and a natural number `ρ`.
+The decision threshold `τ` in the paper corresponds to the precommit threshold used in this LIP up to rescaling and the subtle difference that the weight of precommits in the paper has to be strictly more than `τ` but in these specifications it only has to be greater or equal than `precommitThreshold(h)` (as we restrict the range of `precommitThreshold(h)` such that `precommitThreshold(h) / aggregateBFTWeight(h) > 1/3`).
+Similarly, we choose `prevoteThreshold(h)` such that `prevoteThreshold(h) / aggregateBFTWeight(h) > 2/3`.
+
+The parameter `ρ`, which determines how many blocks a block can imply prevotes for, is given by `3*LSK_BFT_BATCH_SIZE` in the specifications in this LIP, as also in LIP 0014.
+Here `LSK_BFT_BATCH_SIZE` is a constant of the  [BFT module LIP][bft-module-lip].
+As the round length in a PoA chain can vary, we require that the maximum possible round length in a PoA chain, given by the constant `MAX_NUM_VALIDATORS` defined in the [Proof-of-Authority LIP][poa-lip], is always at most the constant `LSK_BFT_BATCH_SIZE`.
```

Comment:
```suggestion
As the round length in a PoA chain can vary, we require that the maximum possible round length in a PoA chain, given by the constant `MAX_NUM_VALIDATORS` defined in [LIP 0047][poa-lip], is always at most the constant `LSK_BFT_BATCH_SIZE`.
```

Reply:
changed to "defined in the PoA module" instead